data-reconciliation-app/src/services/authentication-service.ts

```diff
@@ -0,0 +1,68 @@
+import * as ccfapp from "@microsoft/ccf-app";
+import { ccf } from "@microsoft/ccf-app/global";
+import { User } from "../models/user";
+import { ServiceResult } from "../utils/service-result";
+
+interface Caller {
+  id: string;
+}
+
+export interface IAuthenticationService {
+  getCallerId(request: ccfapp.Request<any>): ServiceResult<User>;
+  isUser(userId: string): ServiceResult<boolean>;
+  isMember(memberId: string): ServiceResult<boolean>;
+}
+
+// Authentication Service
+export class AuthenticationService implements IAuthenticationService {
+  // get caller id
+  // Note that the following way of getting caller ID doesn't work for 'jwt' auth policy and 'no_auth' auth policy.
+  public getCallerId(request: ccfapp.Request<any>): ServiceResult<User> {
+    try {
+      const caller = request.caller as unknown as Caller;
+      return ServiceResult.Succeeded(caller.id);
```

Comment:
Caller Id is string. Can we check if `caller.id` is null or undefined before this?